src/http.rs

```diff
@@ -507,14 +549,24 @@ pub type RawHeaderLine = (String, Vec<u8>);
 /// >                ; obsolete line folding
 /// >                ; see Section 3.2.4
 /// > ```
-pub fn read_header<R: Reader>(stream: &mut R) -> HttpResult<Option<RawHeaderLine>> {
+pub fn read_header<R: Read>(stream: &mut R) -> HttpResult<Option<RawHeaderLine>> {
+    macro_rules! byte (
```

Comment:
This whole macro business is honestly horrible, why is this necessary? Can't we add an extension method?